list/group.principal.vue

```diff
@@ -0,0 +1,102 @@
+<script>
+import ResourceTable from '@/components/ResourceTable';
+import Loading from '@/components/Loading';
+import Masthead from '@/components/ResourceList/Masthead';
+import { NORMAN } from '@/config/types';
+import AsyncButton from '@/components/AsyncButton';
+import { applyProducts } from '@/store/type-map';
+import { NAME } from '@/config/product/auth';
+import { MODE, _EDIT } from '@/config/query-params';
+
+export default {
+  components: {
+    AsyncButton, ResourceTable, Masthead, Loading
+  },
+  props: {
+    resource: {
+      type:     String,
+      required: true,
+    },
+
+    schema: {
+      type:     Object,
+      required: true,
+    },
+  },
+  async fetch() {
+    this.rows = await this.$store.dispatch('cluster/findAll', { type: NORMAN.SPOOFED.GROUP_PRINCIPAL }, { root: true }); // See PromptRemove.vue
+  },
+  data() {
+    return {
+      rows:           null,
+      assignLocation:  {
+        // TODO: Q
+        path:   `/c/local/${ NAME }/${ NORMAN.SPOOFED.GROUP_PRINCIPAL }/assign-edit`,
+        query: { [MODE]: _EDIT }
+      }
+    };
+  },
+  computed: {
+    hasRows() {
+      return this.rows?.length > 0;
+    }
+  },
+  methods: {
+    async refreshGroupMemberships(buttonDone) {
+      try {
+        await this.$store.dispatch('rancher/request', {
+          url:           '/v3/users?action=refreshauthprovideraccess',
+          method:        'post',
+          data:          { },
+        });
+
+        // In SPA this is not needed. In SSR I think this runs client side... where this has not been called (not sure how not...)
+        // If this is not here... when cluster/findAll is dispatched... we fail to find the spoofed type's getInstance fn as it hasn't been
+        // registered yet
+        await applyProducts(this.$store);
```

Comment:
It doesn't get called again because applyProducts puts stuff into the store and the state of the store is transferred to the client.. as JSON, which can't represent functions.  So you didn't create this situation, but the store should never get functions put into it.

Reply:
I'm royally confused now. The root cause of failure was an empty `instanceMethods` const in `type-map.js` which I don't think is included in the official state object? Had assumed that somehow in the client/server split there were two different version of this. Are there any good guides on how this works I can catch up on?

Reply:
I've removed my conjecture from the comment. Might need some help from your side if we need to debug this further. It should be a no-op if already ran, so safe if not nice.